config/routing.go

```diff
@@ -164,10 +164,11 @@ type ComposableRouterParams struct {
 }
 
 type ConfigRouter struct {
-	RouterName   string
-	Timeout      Duration
-	IgnoreErrors bool
-	ExecuteAfter *OptionalDuration `json:",omitempty"`
+	RouterName                     string
+	Timeout                        Duration
+	IgnoreErrors                   bool
+	DoNotWaitForStreamingResponses bool
```

Comment:
Is this the right flag/default and how ready are we to expose this to users (even though it is experimental and subject to change)?

We should consider what we want the defaults to be here (even if they're not all handled now). As I understand it the situation is:

For example for IPNS we have 3 routing systems:
- LAN DHT
- WAN DHT
- IPNS over PubSub

And what we want is:
- If the WAN DHT comes back with a result -> we're done
- If IPNS over PubSub comes back with a result -> we're done
- If the LAN DHT comes back with a result AND the WAN DHT comes back with a failure -> we're done
  - Not sure if this is still an issue since I wasn't around during the routing refactor, but I think so https://github.com/libp2p/go-libp2p-kad-dht/issues/692 
- Otherwise keep waiting until the timeout

However, for provider records we probably want to just keep fetching until we don't need to anymore. The difference being that with provider records getting more is always just advisory and we can keep getting more whereas with IPNS records we frequently just have one chance to choose a record to process for the user.

Note: Peer records are a bit like both but for the most part are more like provider records because we generally end up wanting to connect to the peer itself rather than just asking for what some system thinks that peer's addresses are.

Overall, if this is still not right but is fixing a regression then we shouldn't expose this to users via the config yet.

Reply:
it is not way to confusing, removed from customisation (it's always set now)